tests/benchmark/crud_test.go

```diff
@@ -0,0 +1,895 @@
+// limitations under the License.
+
+package benchmark
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/apigee/registry/connection"
+	"github.com/apigee/registry/rpc"
+	"google.golang.org/genproto/protobuf/field_mask"
+)
+
+func getRootResource() string {
+	registryProject, valueExists := os.LookupEnv("REGISTRY_PROJECT_IDENTIFIER")
+	if valueExists {
+		return fmt.Sprintf("projects/%s/locations/global", registryProject)
+	}
+	return "projects/bench/locations/global"
+}
+
+func getNumberOfVersions() int {
+	strNumberOfVersions, exists := os.LookupEnv("REGISTRY_VERSION_COUNT")
+	numberOfVersions, err := strconv.Atoi(strNumberOfVersions)
+	if err != nil || !exists {
+		numberOfVersions = 1
+	}
+	return numberOfVersions
+}
+
+func getApiName(i int) string {
+	apiNamePrefix, valueExists := os.LookupEnv("API_NAME_PREFIX")
+	if !valueExists {
+		apiNamePrefix = "benchtest"
+	}
+	apiNameStartOffset, valueExists := os.LookupEnv("API_NAME_START_OFFSET")
+	if valueExists {
+		intVar, _ := strconv.Atoi(apiNameStartOffset)
+		i += intVar
+	}
+
+	return fmt.Sprintf("%s-%d", apiNamePrefix, i)
+}
+
+var rootResource = getRootResource()
+
+func readAndGZipFile(filename string) (*bytes.Buffer, error) {
+	fileBytes, _ := ioutil.ReadFile(filename)
+	var buf bytes.Buffer
+	zw, _ := gzip.NewWriterLevel(&buf, gzip.BestCompression)
+	_, err := zw.Write(fileBytes)
+	if err != nil {
+		return nil, err
+	}
+	if err := zw.Close(); err != nil {
+		return nil, err
+	}
+	return &buf, nil
+}
+
+func BenchmarkCreateApi(b *testing.B) {
+	ctx := context.Background()
+	gapicClient, err := connection.NewClient(ctx)
+	if err != nil {
+		b.Fatalf("Setup: Failed to create client: %s", err)
+	}
+	client := gapicClient.GrpcClient()
+
+	for i := 1; i <= b.N; i++ {
+		if i%1000 == 0 {
+			//Sleep 10s every 1000 iterations to reduce server load
+			time.Sleep(10 * time.Second)
+		}
+
+		apiId := getApiName(i)
+		req := &rpc.CreateApiRequest{
+			Parent: rootResource,
+			ApiId:  apiId,
+			Api: &rpc.Api{
+				DisplayName: apiId,
+				Description: fmt.Sprintf("Description for %s", apiId),
+			},
+		}
+		b.StartTimer()
+		_, err := client.CreateApi(ctx, req)
+		b.StopTimer()
+		/**
+		Go Benchmarking always runs all the benchmark tests once before it runs
+		for the duration specified. So the first row always returns error.
+		*/
+		if i > 1 && err != nil {
+			b.Errorf("CreateApi(%+v) returned unexpected error: %s", req, err)
+		}
+	}
+}
+
+func BenchmarkCreateApiVersion(b *testing.B) {
+	ctx := context.Background()
+	gapicClient, err := connection.NewClient(ctx)
+	if err != nil {
+		b.Fatalf("Setup: Failed to create client: %s", err)
+	}
+	client := gapicClient.GrpcClient()
+
+	b.StopTimer()
+	b.ResetTimer()
+
+	for i := 1; i <= b.N; i++ {
+		if i%1000 == 0 {
+			//Sleep 10s every 1000 iterations to reduce server load
+			time.Sleep(10 * time.Second)
+		}
```

Comment:
Why do we need to reduce server load? Why does it need 10s to do that, as opposed to something much quicker like 1s, 100ms, etc?

Reply:
I found the script would error out if I ran it to create 2000 APIs. 
Needed the pause to finish creates sucessfully.